sbor/src/decode.rs

```diff
@@ -113,13 +105,24 @@ impl<'de, X: CustomTypeId> Decoder<'de, X> {
         Ok(slice)
     }
 
-    pub fn check_type_id(&mut self, expected: SborTypeId<X>) -> Result<(), DecodeError> {
-        let ty = self.read_type_id()?;
-        if ty != expected {
-            return Err(DecodeError::UnexpectedTypeId(ty.as_u8()));
+    pub fn check_preloaded_type_id(
+        &self,
+        type_id: SborTypeId<X>,
+        expected: SborTypeId<X>,
+    ) -> Result<SborTypeId<X>, DecodeError> {
+        if type_id == expected {
+            Ok(type_id)
+        } else {
+            Err(DecodeError::UnexpectedTypeId {
+                actual: type_id.as_u8(),
+                expected: expected.as_u8(),
+            })
         }
+    }
 
-        Ok(())
+    pub fn check_type_id(&mut self, expected: SborTypeId<X>) -> Result<SborTypeId<X>, DecodeError> {
```

Comment:
I find this method as written slightly confusingly named - as it does a read as well as a check - unlike the method above?

Perhaps `read_and_check_type_id`? and the method below `read_and_check_size`?

Reply:
Yes!